jobs/AnalyzeResults.php

```diff
@@ -0,0 +1,136 @@
+<?php
+
+namespace MediaSearchSignalTest\Jobs;
+
+use mysqli;
+
+class AnalyzeResults {
+
+    private $db;
+    private $searchId;
+    private $out;
+
+    public function __construct( array $config, int $searchId ) {
+        $this->db = new mysqli( $config['db']['host'], $config['client']['user'],
+            $config['client']['password'], $config['db']['dbname'] );
+        if ( $this->db->connect_error ) {
+            die('DB connection Error (' . $this->db->connect_errno . ') '
+                . $this->db->connect_error);
+        }
+        $this->out = fopen(
+            __DIR__ . '/../out/AnalyzeResults_' . $searchId . '.csv',
+            'w'
+        );
+        $this->searchId = $searchId;
+    }
+
+    public function __destruct() {
+        fclose( $this->out );
+    }
+
+    public function run() {
+        $resultsets = $this->db->query(
+            'select id,term from resultset where searchId = ' . intval( $this->searchId )
+        )->fetch_all( MYSQLI_ASSOC );
+        if ( count( $resultsets ) == 0 ) {
+            die( "ERROR: no results found with for search id " . $this->searchId . "\n");
+        }
+
+        fwrite(
+            $this->out,
+            "Term,f1Score\n"
+        );
+
+        $f1Scores = $precisionTop30s = [];
+        foreach ( $resultsets as $resultset ) {
+            $f1Scores[$resultset['term']] = $this->calculateF1Score(
+                $resultset['id'], $resultset['term'] );
+            $precisionTop30s[$resultset['term']] = $this->calculatePrecisionTop30(
+                $resultset['id'], $resultset['term'] );
+            fwrite( $this->out,
+                $resultset['term'] . "," .
+                $f1Scores[$resultset['term']] . "," .
+                $precisionTop30s[$resultset['term']] . "\n" );
+        }
+        fwrite(
+            $this->out,
+            "ARITHMETIC MEAN," .
```

Comment:
Given how unevenly distributed out labeled data is, it might make sense to aggregate results differently.
We're dealing with sampled data here, and with the current mean, there's a significant difference between how much value we assign to 1 classified file vs another (depending on how many other labels things there are for the same search term)

In short:
search term A = 1 labeled file
search term B = 15 labeled files
both f1scores will contribute to the overall f1score just as much, even though the data for search term B is pretty damn solid and A is almost negligible. Every labeled file should have as much impact on the global score.

See http://rushdishams.blogspot.com/2011/08/micro-and-macro-average-of-precision.html